pkg/config/kpconfig.go

```diff
@@ -1,8 +1,150 @@
 package config
 
-import corev1 "k8s.io/api/core/v1"
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/vmware-tanzu/kpack-cli/pkg/k8s"
+)
+
+const (
+	kpNamespace                         = "kpack"
+	kpConfigMapName                     = "kp-config"
+	canonicalRepositoryKey              = "canonical.repository"
+	canonicalServiceAccountNameKey      = "canonical.repository.serviceaccount"
+	canonicalServiceAccountNamespaceKey = "canonical.repository.serviceaccount.namespace"
+)
 
 type KpConfig struct {
-	CanonicalRepository string
-	ServiceAccount      corev1.ObjectReference
+	canonicalRepository string
+	serviceAccount      corev1.ObjectReference
+}
+
+func NewKpConfig(canonicalRepository string, serviceAccount corev1.ObjectReference) KpConfig {
+	return KpConfig{
+		canonicalRepository: canonicalRepository,
+		serviceAccount:      serviceAccount,
+	}
+}
+
+func (c KpConfig) CanonicalRepository() (string, error) {
+	if c.canonicalRepository == "" {
+		return "", errors.Errorf("key %q not found in configmap %q", canonicalRepositoryKey, kpConfigMapName)
+	}
+
+	return c.canonicalRepository, nil
+}
+
+func (c KpConfig) ServiceAccount() (corev1.ObjectReference, error) {
+	if c.serviceAccount.Name == "" {
+		return corev1.ObjectReference{}, errors.Errorf("key %q not found in configmap %q", canonicalServiceAccountNameKey, kpConfigMapName)
+	}
+
+	return c.serviceAccount, nil
+}
+
+type Helper interface {
+	GetKpConfig(ctx context.Context) (KpConfig, error)
+	SetKpConfig(ctx context.Context, config KpConfig) error
+}
+
+type defaultConfigHelper struct {
+	cs k8s.ClientSet
+}
+
+func NewDefaultConfigHelper(cs k8s.ClientSet) Helper {
```

Comment:
Would provider be a more consistent name with our existing terminology?

Reply:
i would be open to that, i kept the name as it was before but i do think provider make s more sense